modules/data-entry/src/main/frontend/src/questionnaire/Section.jsx

```diff
@@ -0,0 +1,86 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Grid, Typography, withStyles } from "@material-ui/core";
+import uuidv4 from "uuid/v4";
+
+import FormEntry, { ENTRY_TYPES } from "./FormEntry";
+import QuestionnaireStyle, { FORM_ENTRY_CONTAINER_PROPS } from "./QuestionnaireStyle";
+
+// The heading levels that @material-ui supports
+const MAX_HEADING_LEVEL = 6;
+const MIN_HEADING_LEVEL = 3;
+
+/**
+ * Component that consists of a few sections, and optionally has some criteria to its display
```

Comment:
Doesn't seem right...

```suggestion
 * Component responsible for displaying a (sub)section of a questionnaire, consisting of a title, a description,
 * a list of questions and/or subsections, and may have some conditions to its display.
```